spring-rabbit/src/main/java/org/springframework/amqp/rabbit/logback/AmqpAppender.java

```diff
@@ -435,8 +556,60 @@ public void start() {
 	}
 
 	/**
+	 * Create the {@link ConnectionFactory}.
+	 *
+	 * @return a {@link ConnectionFactory}.
+	 */
+	protected ConnectionFactory createRabbitConnectionFactory() {
+		RabbitConnectionFactoryBean factoryBean = new RabbitConnectionFactoryBean();
+		configureRabbitConnectionFactory(factoryBean);
+		try {
+			factoryBean.afterPropertiesSet();
+			return factoryBean.getObject();
+		}
+		catch (Exception e) {
+			addError("Failed to create customized Rabbit ConnectionFactory.", e);
+			return null;
```

Comment:
I'm not good in all those logging tricks, but I would like to have some test for the wrong case, e.g.:
````java
@Test
public void testNotFullSslConfiguration() {
	AmqpAppender appender = new AmqpAppender();
	appender.setUseSsl(true);
	appender.setTrustStore("foo");

	appender.start();
}
````
Where I end up with:
```
LOGBACK: No context given for ch.qos.logback.core.pattern.parser.Compiler@3ac42916
LOGBACK: No context given for ch.qos.logback.core.pattern.parser.Compiler@629f0666
LOGBACK: No context given for org.springframework.amqp.rabbit.logback.AmqpAppender[null]

java.lang.IllegalArgumentException: Target ConnectionFactory must not be null

	at org.springframework.util.Assert.notNull(Assert.java:165)
	at org.springframework.amqp.rabbit.connection.AbstractConnectionFactory.<init>(AbstractConnectionFactory.java:76)
	at org.springframework.amqp.rabbit.connection.CachingConnectionFactory.<init>(CachingConnectionFactory.java:230)
	at org.springframework.amqp.rabbit.logback.AmqpAppender.start(AmqpAppender.java:545)
	at org.springframework.amqp.rabbit.logback.AmqpAppenderTest.testNotFullSslConfiguration(AmqpAppenderTest.java:180)
```
I understand that the first three `LOGBACK` warnings are about not full logback configuration.
But the next `must not be null` may cause some confuse in the target application.

With the real config I have something like this:
```
10:47:12,626 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.PatternLayout] for [layout] property
10:47:12,664 |-ERROR in org.springframework.amqp.rabbit.logback.AmqpAppenderIntegrationTests$EnhancedAppender[AMQP] - Failed to create customized Rabbit ConnectionFactory. java.lang.IllegalStateException: keyStore property required
	at java.lang.IllegalStateException: keyStore property required
	at 	at org.springframework.util.Assert.state(Assert.java:640)
	at 	at org.springframework.amqp.rabbit.connection.RabbitConnectionFactoryBean.setUpSSL(RabbitConnectionFactoryBean.java:484)
...
```
but anyway somewhere down the logs I see:
```
10:47:12,671 |-ERROR in ch.qos.logback.core.joran.spi.Interpreter@27:13 - RuntimeException in Action for tag [appender] java.lang.IllegalArgumentException: Target ConnectionFactory must not be null
	at java.lang.IllegalArgumentException: Target ConnectionFactory must not be null
	at 	at org.springframework.util.Assert.notNull(Assert.java:165)
	at 	at org.springframework.amqp.rabbit.connection.AbstractConnectionFactory.<init>(AbstractConnectionFactory.java:76)
	at 	at org.springframework.amqp.rabbit.connection.CachingConnectionFactory.<init>(CachingConnectionFactory.java:230)
	at 	at org.springframework.amqp.rabbit.logback.AmqpAppender.start(AmqpAppender.java:545)
```
The first one is good, but the second should be avoided.
Seems for me simple `if...else` in the `start()` should be fine.

WDYT?